### PR TITLE
Complete covering of protocol handler

### PR DESCRIPTION
**Pull request by AMikhniev**
Unit tests in commits from https://github.com/LuxoftSDL/sdl_core/commit/3a4d18618018d3085f3860fc15550779dbd54f31 to https://github.com/LuxoftSDL/sdl_core/commit/d1f8a991eca75bea9efcc8903772f7f54251975d reviewed in https://github.com/CustomSDL/sdl_panasonic/pull/165
Related to [APPLINK-20092](https://adc.luxoft.com/jira/browse/APPLINK-20092)
Commits https://github.com/LuxoftSDL/sdl_core/commit/485aa0060c14786d6b70505ee1b081e6eb247de0 and https://github.com/LuxoftSDL/sdl_core/commit/4ae48d529663e80a9ada20104ffde4c4a1bb4f4a need to make tests buildable.
Due to requirements of mock and header files placement related to tests mock files moved to correct places.

### CODE REVIEW

**Comment by AMikhniev** [comment]
@VVeremjova @malirod @Kozoriz @OHerasym @AByzhynar please review.

**Comment by AMikhniev** [comment]
@AByzhynar @Kozoriz @VVeremjova @OHerasym @malirod please review after fixing issued. Some comments lost after squashing, which need to remove fixing of merge conflict from PR commits. But changes are done.

**Comment by OHerasym** [comment]
Reviewed

**Comment by AByzhynar** [comment]
Reviewed

**Comment by VVeremjova** [comment]
Reviewed

**Comment by Kozoriz** [comment]
Reviewed

File: src/components/security_manager/test/security_manager_test.cc [hunk 1]
```diff
@@ -101,11 +101,12 @@ class SecurityManagerTest : public ::testing::Test {
...
    * Wrapper for fast emulate recieve SecurityManager::OnMessageReceived
    */
-  void call_OnMessageReceived(const uint8_t* const data, uint32_t dataSize,
+  void call_OnMessageReceived(const uint8_t* const data,
+                              uint32_t dataSize,
                               const ServiceType serviceType) {
     const ::protocol_handler::RawMessagePtr rawMessagePtr(
-        new ::protocol_handler::RawMessage(key, protocolVersion, data, dataSize,
-                                           serviceType));
+        new ::protocol_handler::RawMessage(
```
Comment (malirod) [inline src/components/security_manager/test/security_manager_test.cc]: @AMikhniev , double check all places in this PR where you use `new`. Replace with `MakeShared` where possible. In this particular place the call of `new` can be avoided.

Comment (AMikhniev) [inline src/components/security_manager/test/security_manager_test.cc]: @malirod see https://github.com/LuxoftSDL/sdl_core/commit/9bc0e12bc4d26d165ebce1b92fdac612dab0924f

**Comment by malirod** [comment]
Reviewed.

**Pull request merged**